go/libkb/sig_chain.go

```diff
@@ -665,7 +699,81 @@ func (sc *SigChain) VerifySigsAndComputeKeys(ctx context.Context, eldest keybase
 	// the id_table.  See LoadUser in user.go and
 	// https://github.com/keybase/go/issues/43
 
-	return
+	return cached, len(links), nil
+}
+
+func reverseListOfChainLinks(arr []ChainLinks) {
+	for i, j := 0, len(arr)-1; i < j; i, j = i+1, j-1 {
+		arr[i], arr[j] = arr[j], arr[i]
+	}
+}
+
+// VerifySigsAndComputeKeys iterates over all potentially all incarnations of the user, trying to compute
+// multiple subchains. It returns (bool, error), where bool is true if the load hit the cache, and false othewise.
+func (sc *SigChain) VerifySigsAndComputeKeys(ctx context.Context, eldest keybase1.KID, ckf *ComputedKeyFamily, loadAllSubchains bool) (bool, error) {
+	// First consume the currently active sigchain.
+	cached, numLinksConsumed, err := sc.verifySigsAndComputeKeysCurrent(ctx, eldest, ckf)
+	if !loadAllSubchains || err != nil || ckf.kf == nil {
+		return cached, err
+	}
+
+	allCached := cached
+
+	// ignore error here, since it shouldn't kill the overall load if historical subchains don't run
+	// correctly.
+	cached, _ = sc.verifySigsAndComputeKeysHistorical(ctx, numLinksConsumed, *ckf.kf)
```

Comment:
Since this is the only caller, maybe the right thing to do is to have the function not return an error at all?